src/cirrus/lib2/statedb.py

```diff
@@ -490,14 +490,15 @@ def query(
             index = "state_updated"
             expr = expr & Key(index).begins_with(state)
 
-        keys = ["collections_workflow", "itemids"]
+        exclusive_start_key_filters = ["collections_workflow", "itemids"]
 
         if index:
             kwargs["IndexName"] = index
+            exclusive_start_key_filters.append(index)
```

Comment:
This was the underlying issue affecting `api` and `rerun`. In short, an older `moto` version broke the dynamo tests when `index` was included in the `ExclusiveStartKey`, so I ended up removing this line to get those tests to work because I thought this was a bug in the code here. Turns out it was a bug in `moto`; a newer moto version fixes this (though still has different behavior than dynamo, in that omitting `index` still works with `moto` but not with dynamo).

Adding this line back in resolves the problem. All other changes in this file are just for clarity/logging.